core/model/src/main/java/org/activityinfo/model/type/number/QuantityType.java

```diff
@@ -47,17 +63,60 @@ public QuantityType(String units) {
         this.units = units;
     }
 
+    public QuantityType(String units, String aggregation) {
+        this(units);
+        this.aggregation = Aggregation.valueOf(aggregation);
+    }
+
+    public QuantityType(String units, Aggregation aggregation) {
+        this(units);
+        this.aggregation = aggregation;
+    }
+
+    private void inputAggregation(int code) {
```

Comment:
Methods that mutate object state should be called "setXXX". You can merge this into setAggregation below or make this a static method of the Aggregation enum called "valueOf".